js/coindcx.js

```diff
@@ -0,0 +1,596 @@
+            'limit': limit,
+        };
+        const response = await this.privatePostExchangeV1OrdersTradeHistory (this.extend (request, params));
+        return this.parseTrades (response, undefined, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        const timestamp = this.safeInteger2 (trade, 't', 'timestamp');
+        let symbol = undefined;
+        if (market === undefined) {
+            const marketId = this.safeString2 (trade, 's', 'symbol');
+            market = this.safeValue (this.markets_by_id, marketId);
+        }
+        if (market !== undefined) {
+            symbol = market['symbol'];
+        }
+        let takerOrMaker = undefined;
+        if ('m' in trade) {
+            takerOrMaker = trade['m'] ? 'maker' : 'taker';
+        }
+        const price = this.safeFloat2 (trade, 'p', 'price');
+        const amount = this.safeFloat2 (trade, 'q', 'quantity');
+        return {
+            'id': this.safeString (trade, 'id'),
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'order': undefined,
+            'type': undefined,
+            'takerOrMaker': takerOrMaker,
+            'side': this.safeString (trade, 'side'),
+            'price': price,
+            'amount': amount,
+            'cost': price * amount,
+            'fee': this.safeFloat (trade, 'fee_amount'),
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?shell#order-book
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
+            throw new ExchangeError (this.id + ' has no symbol ' + symbol);
+        }
+        const market = this.market (symbol);
+        const marketInfo = this.safeValue (market, 'info');
+        const coindcxPair = this.safeString (marketInfo, 'pair');
+        if (coindcxPair === undefined) {
+            throw new ExchangeError (this.id + ' has no pair (look at market\'s info) value for ' + symbol);
+        }
+        const request = {
+            'pair': coindcxPair,
+        };
+        const response = await this.publicGetMarketDataOrderbook (this.extend (request, params));
+        return this.parseOrderBookData (response);
+    }
+
+    parseOrderBookData (orderBook) {
+        const bids = this.safeValue (orderBook, 'bids', {});
+        const asks = this.safeValue (orderBook, 'asks', {});
+        return {
+            'bids': this.sortBy (this.parseBidAskData (bids), 0, true),
+            'asks': this.sortBy (this.parseBidAskData (asks), 0),
+            'timestamp': undefined,
+            'datetime': undefined,
+            'nonce': undefined,
+        };
+    }
+
+    parseBidAskData (bidsOrAsks) {
```

Comment:
This method is also redundant, see the base `parseOrderBook` method and how it is called in all other exchanges.